syntax/repr.rs

```diff
@@ -0,0 +1,45 @@
+use crate::syntax::Atom::{self, *};
+use syn::parse::{Error, Parse, ParseStream, Result};
+use syn::{Ident, LitInt};
+
+#[derive(Copy, Clone, PartialEq)]
+pub enum Repr {
+    Align(u32),
+    Atom(Atom),
+}
+
+impl Parse for Repr {
+    fn parse(input: ParseStream) -> Result<Self> {
+        let begin = input.cursor();
+        let ident: Ident = input.parse()?;
+        if let Some(atom) = Atom::from(&ident) {
+            match atom {
+                U8 | U16 | U32 | U64 | Usize | I8 | I16 | I32 | I64 | Isize if input.is_empty() => {
+                    return Ok(Repr::Atom(atom));
+                }
+                _ => {}
+            }
+        } else if ident == "align" {
+            let content;
+            syn::parenthesized!(content in input);
+            let alignment: u32 = content.parse::<LitInt>()?.base10_parse()?;
```

Comment:
This would accept a suffixed integer literal such as `#[repr(align(2int))]`. That should not be allowed.